### PR TITLE
Added backend for svg in UIToolkit renderer

### PR DESCRIPTION
**Pull request by Muchaszewski**
Added svg backend support for UIToolkit renderer. There seems to be guid bleed from adding the `UnityEngine.VectorGraphics` package to UIToolkit in multiple packages

### CODE REVIEW

File: Runtime/Frameworks/UIToolkit/VisualElements/SvgElement.cs [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+using Object = UnityEngine.Object;
+
+#if REACT_VECTOR_GRAPHICS
+using Unity.VectorGraphics;
+#endif
+
+public class SvgElement : Image
+{
+    private static readonly string styleName = "TabButtonStyles";
+    private static readonly string UxmlName = "Svg";
```
Comment (KurtGokhan) [inline Runtime/Frameworks/UIToolkit/VisualElements/SvgElement.cs:16]: Is Uxml stuff actually needed? See my other comment.

File: Runtime/Frameworks/UIToolkit/VisualElements/SvgElement.cs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+using System.Reflection;
+using UnityEngine;
+using UnityEngine.UIElements;
+using Object = UnityEngine.Object;
+
+#if REACT_VECTOR_GRAPHICS
+using Unity.VectorGraphics;
+#endif
+
+public class SvgElement : Image
```
Comment (KurtGokhan) [inline Runtime/Frameworks/UIToolkit/VisualElements/SvgElement.cs:13]: Namespace should be `ReactUnity.UIToolkit`

File: Runtime/Frameworks/UIToolkit/VisualElements/SvgElement.cs [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    ///     Delegate of MakeVector method
+    /// </summary>
+    /// <remarks>This method is not supported on iOS and Web, and needs to fallback to standard reflection</remarks>
+    delegate void MakeVectorDelegate(List<VectorUtils.Geometry> geometry, uint gradientResolution, out Object asset,
+        out Texture2D texture2D);
+#endif
+#endif
+
+    [SerializeField]
+    private string _rawSvg;
```
Comment (KurtGokhan) [inline Runtime/Frameworks/UIToolkit/VisualElements/SvgElement.cs:35]: Should be `rawSvg` according to this project's naming conventions so far. We don't use `_` at the start of private members. Only camelCase.

File: Runtime/Frameworks/UIToolkit/Components/SvgComponent.cs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                    InnerContent = text.text;
+                    FireEvent("onLoad", new { type = "load" });
+                });
+        }
+
+        protected void RefreshValue()
+        {
+#if !REACT_VECTOR_GRAPHICS
+            Debug.LogError(
+                "UnityEngine.VectorGraphics is not enabled, enable this package through REACT_VECTOR_GRAPHICS to allow for svg");
```
Comment (KurtGokhan) [inline Runtime/Frameworks/UIToolkit/Components/SvgComponent.cs:100]: Message is not accurate. Can be `Unity.VectorGraphics module is required to use SVG components`

File: Runtime/Frameworks/UIToolkit/Components/SvgComponent.cs [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+                RefreshValue();
+            }
+        }
+
+        public object Source
+        {
+            get
+            {
+                return source;
+            }
```
Comment (KurtGokhan) [inline Runtime/Frameworks/UIToolkit/Components/SvgComponent.cs:57]: Let's write as `get => source;` for brevity. Same for other props.

**Review by KurtGokhan — CHANGES_REQUESTED** [review]

File: Runtime/Frameworks/UIToolkit/VisualElements/SvgElement.cs [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+#endif
+#endif
+
+
+    public string RawSvg
+    {
+        get
+        {
+            return _rawSvg;
+        }
```
[the remark this answers is not in the document]
Comment (Muchaszewski) [reply Runtime/Frameworks/UIToolkit/VisualElements/SvgElement.cs:79]: It's a good practice to keep properties consistent with syntax within the property block. Eg:
```
// Ok
string Foo
{
   get => _foo;
   set => _foo = value;
}
// Ok
string Foo
{
   get { return _foo; }
   set { _foo = value; }
}
// Not Ok
string Foo
{
   get => _foo;
   set { _foo = value; more code... }
}
```

**Comment by KurtGokhan** [comment]
Looks good. Will merge after I test it

**Review by KurtGokhan — APPROVED** [review]

**Comment by KurtGokhan** [comment]
The asmdef changes seemed extra, so I reverted them.

**Pull request merged**